### PR TITLE
Fix(rome_js_formatter):regex flag order

### LINKED ISSUE #2445: Sort regular expression flags
Prettier sorts the regular expression flags and so should Rome's formatter [Playground](https://play.rome.tools/?lineWidth=80&indentStyle=space&quoteStyle=double&indentWidth=2&typescript=false&jsx=true&sourceType=module#LwBbAC8AXQBcAC8AXAB1ADAAYQBCAGMALwBtAGcAaQA7AAoA)

## Input 

```ts
/[/]\/\u0aBc/mgi;
```

## Prettier

```ts
/[/]\/\u0aBc/gim;
```

## Rome

```ts
/[/]\/\u0aBc/mgi;
```

## Expected

Rome's formatting to match Prettier's

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
## Summary
Resolve #2445

## Test Plan

Snapshot test result has been aligned to related issue, 
now regex literal flags should be sorted correctly

### CODE REVIEW

File: crates/rome_js_formatter/src/js/expressions/regex_literal_expression.rs [hunk 1]
```diff
@@ -1,13 +1,40 @@
...
 use rome_js_syntax::JsRegexLiteralExpressionFields;
 
 impl FormatNode for JsRegexLiteralExpression {
     fn format_fields(&self, formatter: &Formatter) -> FormatResult<FormatElement> {
         let JsRegexLiteralExpressionFields { value_token } = self.as_fields();
+        let value_token = value_token?;
+        let trimmed_raw_string = value_token.text_trimmed();
+        // find end slash, so we could split our regex literal to two part `body`: raw_string[0..end_slash_pos + 1] and `flags`: raw_string[end_slash_pos + 1..]
+        // reference https://tc39.es/ecma262/#prod-RegularExpressionLiteral
+        let end_slash_pos = trimmed_raw_string
```
Comment (ematipico) [inline crates/rome_js_formatter/src/js/expressions/regex_literal_expression.rs:16]: We can bail early if the string ends with `/`. If so, we don't have any flags and we can just return without even calling `rfind` :)

File: crates/rome_js_formatter/src/js/expressions/regex_literal_expression.rs [hunk 2]
```diff
@@ -1,13 +1,40 @@
...
+            .rfind('/')
+            .ok_or(FormatError::MissingRequiredChild)?;
```
Comment (ematipico) [inline crates/rome_js_formatter/src/js/expressions/regex_literal_expression.rs:18]: Do you think we can hit this error? Maybe using `unwrap` and adding a `// SAFETY` comment might work.

**Review by ematipico — APPROVED** [review]
Do you mind to create a new test inside our test suite with exact same text? The reason I ask this is because a tomorrow we might not have the prettier testing suite, and relying on ours is always good.

**Comment by IWANABETHATGUY** [comment]
!bench_formatter

**Comment by IWANABETHATGUY** [comment]
@ematipico I guess, this pr is ready to merge

**Review by ematipico — APPROVED** [review]

**Pull request merged**